GeneratedRegexPatterns/MediumConfidenceSecurityModels.json

```diff
@@ -20,10 +20,11 @@
     "DetectionMetadata": "HighEntropy, MediumConfidence"
   },
   {
-    "Pattern": "https?:\\/\\/(?:[^:@]+):(?<refine>[^:@?]+)@",
+    "Pattern": "(ftps?|https?):\\/\\/(?:[^:@]+):(?<refine>[^:@?]+)@",
```

Comment:
>ftps [](http://example.com/codeflow?start=17&length=4)

Somehow our URL credentials pattern got out of sync with ADO tests. I believe this happened because we had to replace this pattern entirely because the original ADO form contains back-tracking, which we do not support.

Reply:
Keep in mind that I am a newbie when it comes to regexes, but I wonder if we could have some central repository of secret regexes that allows for different flavors of regexes for the same purpose to be contributed.

Then they could all be run through the same tests and we could comment on why this one is slightly different from that one, etc.

Reply:
Sure thing. So I believe in January we will be looking at two related things: 1) additional investments in this library and its factoring, 2) a new general plugin model for our larger e2e tool. That's a great context for us to discuss the topic you've raised and I'll be sure that we cover it. We can talk offline, perhaps early in 2025. :)